### PR TITLE
KPI: add new placeholder build to handle processing of Allure data.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Proposed Changes

This PR adds a new placeholder build that is intended to process the generated Allure results.
This PR also fleshes out the KPIDashboardTests experimental build configuration to reflect how a future Pre-Release Tests build may look like.

Key changes:
- addition of params, steps and artifact rules for the KPIDashboardTests.
- addition of new project "Calypso Dashboard" with a new build configuration `CalypsoPreReleaseDashboard` which will be tasked with processing and uploading of the report.
- new secrets to interact with AWS S3 bucket.

#### Testing Instructions
Ensure the following build configurations are passing:
  - [x] Code Style
  - [x] Unit Tests

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

**Comment by worldomonation** [comment]
FYI, in the process of making this PR the TeamCity configuration got messed up: context in pMz3w-gcN-p2.

File: .teamcity/settings.kts [hunk 1]
```diff
@@ -82,6 +83,12 @@ project {
 		password("E2E_SECRETS_ENCRYPTION_KEY_AUG_22_22", "credentialsJSON:e74f4821-2033-4ef2-b9f9-95c1d7c3898a", display = ParameterDisplay.HIDDEN)
 		password("CONFIG_E2E_ENCRYPTION_KEY_LEGACY", "credentialsJSON:819c139c-90a1-4803-8367-00e5aa5fdb07", display = ParameterDisplay.HIDDEN)
 		password("E2E_SECRETS_ENCRYPTION_KEY_CURRENT", "%E2E_SECRETS_ENCRYPTION_KEY_AUG_22_22%", display = ParameterDisplay.HIDDEN)
+
+		// Calypso dashboard AWS secrets for S3 bucket.
+		password("CALYPSO_E2E_DASHBOARD_AWS_S3_ACCESS_KEY_ID", "credentialsJSON:1f324549-3795-43e5-a8c2-fb81d6e7c15d", display = ParameterDisplay.HIDDEN)
+		password("CALYPSO_E2E_DASHBOARD_AWS_S3_SECRET_ACCESS_KEY", "credentialsJSON:782b4bde-b73d-4326-9970-5a79251bdf07", display = ParameterDisplay.HIDDEN)
+		password("CALYPSO_E2E_DASHBOARD_AWS_S3_ROOT", "s3://a8c-calypso-e2e-reports")
```
Comment (dpasque) [inline .teamcity/settings.kts]: **Question:** Is this a sensitive value? I figure if it's not coming off of the credentials JSON, we probably don't need to use `password`, right? And could just use `param` or `text`?

File: .teamcity/_self/projects/CalypsoDashboard.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import _self.bashNodeScript
+
+import jetbrains.buildServer.configs.kotlin.v2019_2.Project
+import jetbrains.buildServer.configs.kotlin.v2019_2.Dependencies
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildStep
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildSteps
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.finishBuildTrigger
+
+object CalypsoDashboard : Project({
```
Comment (dpasque) [inline .teamcity/_self/projects/CalypsoDashboard.kt]: **Suggestion/Question:** Okay, kind of playing devil's advocate here... But should we move this back into the `WebApp` project for now?

I really appreciate you trying to break up the massive behemoth that the `WebApp` project has become... We could certainly use some more granular modularization.

And at the same time, this distinction feels a little weird to me. This is one of the only Calypso related things that isn't in the `WebApp` project, and it is tightly coupled to a build in the `WebApp` project.

What if we just put this in WebApp for now, and do a separate dedicated PR to try to break `WebApp` up into multiple steps?

File: .teamcity/_self/projects/WebApp.kt [hunk 3]
```diff
@@ -874,18 +874,124 @@ object HelpCentreE2ETests : E2EBuildType(
...
+				find test/e2e/allure-results -name '*.json' -print0 | xargs -r -0 tar cvfz allure-results.tgz
+			""".trimIndent()
+			dockerImage = "%docker_image_e2e%"
+		}
+	}
+
+	features {
+		perfmon {
+		}
+		commitStatusPublisher {
```
Comment (dpasque) [inline .teamcity/_self/projects/WebApp.kt]: **Suggestion:** We can remove this right? We don't need to send this info back to GitHub in any way, do we?

**Review by dpasque — APPROVED** [review]
Approving! I left a few questions and suggestions, but they aren't blocking and I don't think it will need re-review.

From what I could tell, I wasn't able to test the new sub-project in TeamCity prior to merge, but I did run the updated build.

File: .teamcity/settings.kts (hunk 1 above)
Comment (worldomonation) [reply .teamcity/settings.kts]: Right, I had intended to keep it a secret value initially, but it seemed a little unnecessary. Let's switch it back to a text.

File: .teamcity/_self/projects/CalypsoDashboard.kt (hunk 2 above)
Comment (worldomonation) [reply .teamcity/_self/projects/CalypsoDashboard.kt]: Done - merged the new build configuration back into WebApp.kt.

File: .teamcity/_self/projects/WebApp.kt (hunk 3 above)
Comment (worldomonation) [reply .teamcity/_self/projects/WebApp.kt]: Done - removed the commit publisher.

**Pull request merged**